modules/dxp/apps/app-builder-workflow/app-builder-workflow-web/src/main/resources/META-INF/resources/js/pages/apps/edit/configReducer.es.js

```diff
@@ -349,6 +367,14 @@ export default (state, action) => {
 			return {...state, currentStep};
 		}
 		case UPDATE_STEP_INDEX: {
+			if (state.steps[action.stepIndex].initial === undefined) {
+				state.steps[
+					action.stepIndex
+				].appWorkflowDataLayoutLinks = filterDataLayoutList(
+					state.steps[action.stepIndex].appWorkflowDataLayoutLinks
+				);
+			}
+
```

Comment:
@gabrielpramos What do you think about always call the `REMOVE_STEP_EMPTY_FORM_VIEWS` action before call the `UPDATE_STEP_INDEX` instead of repeat the logic?